extensions/wikia/ArticleComments/classes/ArticleComment.class.php

```diff
@@ -335,13 +334,34 @@ private function loadLastRevision() {
 		return !empty( $this->mLastRevision ) && $this->mLastRevision instanceof Revision;
 	}
 
-	public function parseText( $rawText ) {
+	/**
+	 * @param string $rawText
+	 */
+	public function setRawText( $rawText ) {
+		$this->mRawtext = $rawText;
+
+		// invalidate locally cached value that will be set when parseText() is called
+		$this->mText = false;
+		$this->mHeadItems = false;
+	}
+
+	/**
+	 * Returns the wikitext content of the comment with filtering done by self::removeMetadataTag()
+	 *
+	 * @@return string
+	 */
+	public function getRawText() {
+		return self::removeMetadataTag( $this->mRawtext );
+	}
+
+	/**
+	 * Parse the comment content in a lazy fashion: when either getText() or getHeadItems() is called
+	 */
+	private function parseText() {
 		wfProfileIn( __METHOD__ );
 
 		global $wgEnableParserCache;
 
-		$this->mRawtext = self::removeMetadataTag( $rawText );
-
 		# seriously, WTF?
 		$wgEnableParserCache = false;
```

Comment:
Added a caching around the text and metadata returned by the parser:

```
22.33% 0.253656      9 - ArticleComment::parseText
 0.50% 0.004309      9 - ArticleComment::parseText
```

Profiler data for a simple user message wall page (wth a few messages there) shows a huge improvement.